modules/dnn/src/layers/layers_common.cpp

```diff
@@ -144,14 +144,20 @@ void getStrideAndPadding(const LayerParams &params, std::vector<size_t>& pads_be
 }
 }
 
-void getPoolingKernelParams(const LayerParams &params, std::vector<size_t>& kernel, bool &globalPooling,
+void getPoolingKernelParams(const LayerParams &params, std::vector<size_t>& kernel, std::vector<bool>& globalPooling,
                             std::vector<size_t>& pads_begin, std::vector<size_t>& pads_end,
                             std::vector<size_t>& strides, cv::String &padMode)
 {
-    globalPooling = params.has("global_pooling") &&
-                    params.get<bool>("global_pooling");
-
-    if (globalPooling)
+    bool is_global = params.get<bool>("global_pooling", false);
+    globalPooling = std::vector<bool>(3, is_global);
+    if (params.has("global_pooling_d"))
+        globalPooling[0] = params.get<bool>("global_pooling_d");
+    if (params.has("global_pooling_h"))
+        globalPooling[1] = params.get<bool>("global_pooling_h");
+    if (params.has("global_pooling_w"))
+        globalPooling[2] = params.get<bool>("global_pooling_w");
+
+    if (is_global)
```

Comment:
Due global pooling across depth is not tested yet, to match previous behavior it's better to check 
```cpp
if (globalPooling[1] || globalPooling[2])
```